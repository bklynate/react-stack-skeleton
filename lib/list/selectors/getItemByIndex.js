'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = getItemByIndex;
function getItemByIndex(state, index) {
  var items = state.items;


  for (var i = 0; i < items.length; i++) {
    if (i === index) return items[i];
  }

  throw new Error('Invalid todo list index: ' + index);
}