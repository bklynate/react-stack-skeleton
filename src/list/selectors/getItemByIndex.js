/* @flow */
import type { State } from '../reducers/todoItemsReducer';

export default function getItemByIndex(state: State, index: number): * {
  const { items } = state;

  for(let i=0; i<items.length; i++) {
    if(i === index) return items[i];
  }

  throw new Error(`Invalid todo list index: ${index}`);
}
