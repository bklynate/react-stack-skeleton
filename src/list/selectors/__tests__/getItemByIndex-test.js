import getItemByIndex from '../getItemByIndex';

test('it gets the item, using the given index, from the state', () => {
  const items = [ { value: 'test 0' }, { value: 'test 1' }, { value: 'test 2' } ];
  for(let i=0; i<items.length; i++) {
    const selectedState = getItemByIndex({ items }, i);

    expect(selectedState).toEqual(items[i]);
  }
});

test('it throws an error if the given index is not in the state', () => {
  const items = [ { value: 'test 0' }, { value: 'test 1' }, { value: 'test 2' } ];
  const selectState = () => getItemByIndex({ items }, 5);

  expect(selectState).toThrowErrorMatchingSnapshot();
});
