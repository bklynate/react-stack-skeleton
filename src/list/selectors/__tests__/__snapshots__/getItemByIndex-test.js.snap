exports[`test it throws an error if the given index is not in the state 1`] = `"Invalid todo list index: 5"`;
